akka-javasdk-tests/src/test/java/akkajavasdk/components/workflowentities/TransferWorkflowWithFraudDetection.java

```diff
@@ -69,10 +69,12 @@ public Effect<Message> acceptTransfer() {
       return effects().reply(new Message("transfer not started"));
     } else if (!currentState().accepted() && !currentState().finished()) {
       var withdrawInput = new Withdraw(currentState().transfer().from(), currentState().transfer().amount());
-      return effects()
-          .updateState(currentState().asAccepted())
-          .transitionTo(withdrawStepName, withdrawInput)
-          .thenReply(new Message("transfer accepted"));
+
+      Effect.TransitionalEffect<Void> transitional = effects()
+        .updateState(currentState().asAccepted())
+        .transitionTo(withdrawStepName, withdrawInput);
+
+      return transitional.thenReply(new Message("transfer accepted"));
```

Comment:
Actually we are quite safe.

Here `transitionTo` returns `Effect.Transitional` but we can assign it to `Effect.TransitionalEffect<Void>` as if it was an alias.